packages/app/src/components/ErrorModal.tsx

```diff
@@ -1,12 +1,113 @@
-import { Dialog, DialogTitle, DialogContent, DialogContentText } from "@mui/material"
+import { useState } from "react";
+import { Dialog, DialogTitle, DialogContent, DialogContentText, FormControl, InputLabel, Checkbox, ListItemText, MenuItem, OutlinedInput, Select, SelectChangeEvent, Button } from "@mui/material"
+import { useRecoilValue, useSetRecoilState } from "recoil";
+import { dataSelector, encodedDataAtom } from "../atoms/dataAtom";
+import { CoreUpsetData, process } from "@visdesignlab/upset2-core";
+
+const ITEM_HEIGHT = 48;
+const ITEM_PADDING_TOP = 8;
+const MenuProps = {
+  PaperProps: {
+    style: {
+      maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
+      width: 250,
+    },
+  },
+};
 
 export const ErrorModal = () => {
+    const data = useRecoilValue(dataSelector);
+    const setEncodedDataAtom = useSetRecoilState(encodedDataAtom);
+    const [ encodeList, setEncodeList ] = useState<string[]>([]);
+    let needOneHot = false;
+
+    if (data === null) return null;
+
+    if (Object.values(data.columnTypes).includes('category')) {
+        needOneHot = true;
+    }
+
+    const handleChange = (e: SelectChangeEvent) => {
+        const {
+            target: { value },
+          } = e;
+        setEncodeList(typeof value === 'string' ? value.split(',') : value)
+    }
+
+    const oneHotEncode = (empty?: boolean) => {
+        // close the error window and rerender screen by updating the atom
+        if (empty) {
+            setEncodedDataAtom(null);
+        }
+
+        const newColNames: string[] = []
+        const encodedData: CoreUpsetData = structuredClone(data);
+
+        // get the unique names of every new column to be added. 
+        //    ex: group1_a, group1_b, group2_a, group2_b, group3_c, ....
+        encodeList.forEach((s) => Object.entries(data.items).forEach(([id, row]) => {
+            const names = `${row[s]}`.split(',');
+            if (!(names.length === 1 && names[0] === ""))
```

Comment:
I think this will cause issues with values like `a,,c` and `a,b,` — we need to ignore all spaces. Let's do more testing here

Reply:
![image](https://user-images.githubusercontent.com/35744963/228014893-b30bf868-9a30-4a12-8cb7-cc50acd0ebe8.png)
Looks like it populated correctly. Note the highlighted row and the one below are the `a,b,` and `a,,c` intersections.